kolibri/plugins/registry.py

```diff
@@ -48,14 +67,15 @@ def initialize():
             try:
                 plugin_module = importlib.import_module(app + ".kolibri_plugin")
                 logger.debug("Loaded kolibri plugin: {}".format(app))
+                all_classes = dict([(name, cls) for name, cls in plugin_module.__dict__.items() if isinstance(cls, type)])
                 plugin_classes = []
-                for obj in plugin_module.PLUGINS:
+                for obj in all_classes.values():
```

Comment:
The reason that we used the `PLUGINS` module variable convention is that it turns out inspecting all the names in a module will *also* include names that the module imports. So it was possible to have the *same* plugin class be imported twice or more. Do you avoid that here? Presumably if two different plugins both import the same abstract hook, then in this case it would appear twice and be instantiated below twice, once for each module that imported it. Is that the intended behavior?

It might be better to iterate over all plugin modules and add each hook class to a set before iterating over the set to ensure that it's instantiated below only *once* even if two different modules import it.